Add slightly working rag agents, seemingly fixed sessions stopping unexpectedly

failandimprove1 opened the pull request.

failandimprove1 committed:
add example groupchat with RAG

test version copied from autogen blog

failandimprove1 committed:
Merge branch 'alpha' into feature/rag-agents

failandimprove1 committed:
add gpt-4-turbo model

failandimprove1 committed:
Merge branch 'alpha' into feature/rag-agents

failandimprove1 committed:
Merge remote-tracking branch 'origin' into feature/rag-agents

failandimprove1 committed:
add rag test endpoint to my sandbox endpoints

failandimprove1 committed:
Merge remote-tracking branch 'origin' into feature/rag-agents

failandimprove1 committed:
Merge remote-tracking branch 'origin' into feature/rag-agents

failandimprove1 committed:
remove random unused imports

failandimprove1 committed:
add test rag class to crew

failandimprove1 committed:
start of rag implementation, errors atm

when trying to use the register_for_llm decorator on the
retrieve_content method it breaks, the decorator doesn't seem to work
with that method for some reason. it can't find a "_name" attribute

failandimprove1 committed:
add mega primitive implementation of rag

its super scuffed rn bewary before reading

failandimprove1 committed:
move rag.py to lib

failandimprove1 committed:
gitignore and delete tmp folder

failandimprove1 committed:
format (and make task on rag options nullable)

failandimprove1 committed:
format again lol

failandimprove1 committed:
Merge branch 'alpha' into feature/rag-agents

failandimprove1 committed:
make retrieve_content a method

failandimprove1 committed:
chat result tests

failandimprove1 committed:
Merge branch 'alpha' into feature/rag-agents

failandimprove1 committed:
poetry

failandimprove1 committed:
change to using the passed rag options

also removed the extra 0.01 cost when adding cost margin

failandimprove1 committed:
Merge branch 'alpha' into feature/rag-agents

failandimprove1 committed:
make RagOptions nullable

it is being defaulted in the sessions/run endpoint atm

failandimprove1 committed:
format

failandimprove1 committed:
ignore rag user proxy if use_rag is false (when rag_options is None)

failandimprove1 committed:
Merge branch 'alpha' into feature/rag-agents

failandimprove1 committed:
fixups on bugs etc

failandimprove1 committed:
merge didnt work for some reason

failandimprove1 committed:
got decently working rag and fixed sessions stopping randomly

eksno reviewed (approved):
Congratulations or I'm sorry that happened

failandimprove1 committed:
refactor: remove commented code

This commit removes large blocks of commented code from the files'apps/api/src/lib/rag.py' and 'apps/api/src/routers/sandbox/leon.py'.It also rearranges the import statements in 'apps/api/src/lib/crew.py'.This cleanup improves readability and maintainability of the codebase.

failandimprove1 committed:
refactor: remove unused function registration

This commit removes the unused function registration in the file
'apps/api/src/routers/sandbox/leon.py'. The function 'function_moment'
was registered but never used. This change improves the clarity of the
code and reduces potential confusion.
